crates/sui-framework/sources/display.move

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Defines a Display struct which defines the way an Object
+/// should be displayed. The intention is to keep data as independent
+/// from its display as possible, protecting the development process
+/// and keeping it separate from the ecosystem agreements.
+///
+/// Each of the fields of the Display object should allow for pattern
+/// substitution and filling-in the pieces using the data from the object T.
+///
+/// More entry functions might be added in the future depending on the use cases.
+module sui::display {
+    use sui::publisher::{is_package, Publisher};
+    use sui::tx_context::TxContext;
+    use std::string::{String, utf8};
+    use sui::vec_map::{Self, VecMap};
+    use sui::object::{Self, ID, UID};
+    use sui::transfer;
+    use sui::event;
+    use std::vector;
+
+    /// For when T does not belong to package in PublisherCap.
+    const ENotOwner: u64 = 0;
+
+    /// For when vectors passed into one of the multiple insert functions
+    /// don't match in their lengths.
+    const EVecLengthMismatch: u64 = 1;
+
+    /// The Display object. Defines the way an object should be
+    /// displayed. Display object can only be created and modified with
+    /// a PublisherCap, making sure that the rules are set by the owner
+    /// of the type.
+    ///
+    /// Each of the display properties should support patterns outside
+    /// of the system, making it simpler to customize Display based
+    /// on the property values of an Object.
+    /// ```
+    /// // Example of a display object
+    /// Display<0x...::capy::Capy> {
+    ///  fields:
+    ///    <name, "Capy {{ genes }}">
```

Comment:
I think the `{{` syntax will be confusing without additional explanation. Perhaps we could start with an example using very basic strings, then use this example to demonstrate pattern substitution in the frontend?

Reply:
Thoughts about making it "Capy {genes}" 

That is, a single set of {} for these values. With it being whitespace insensitive inside the brackets. 

We can then do the rust trick of {{ to escape a bracket.